### PR TITLE
Rewrite workflow using reusable actions

### PR DESCRIPTION
**Pull request by archanaserver**
Updating the Ruby test workflow by using shared rubocop and foreman plugin test action:
https://github.com/theforeman/actions#rubocop
https://github.com/theforeman/actions#foreman-plugin-tests

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,84 +1,32 @@
...
-          npm install
-          bundle exec rake webpack:compile
-      - name: Run plugin tests
-        uses: nick-fields/retry@v2
-        with:
-          max_attempts: 5
-          timeout_minutes: 30
-          command: |
-            bundle exec rake --trace test:foreman_dlm
-            bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"
```
Comment (archanaserver) [inline .github/workflows/ci.yml:76]: related discussion https://github.com/theforeman/actions/pull/16

**Comment by laugmanuel** [comment]
LGTM, however I'll wait until the discussion regarding `access_permissions_test` is solved.

**Comment by archanaserver** [comment]
I've temporarily drafted this until the issue at https://github.com/theforeman/actions/pull/16 is resolved

**Comment by laugmanuel** [comment]
I'll do this as part of https://github.com/dm-drogeriemarkt/foreman_dlm/pull/67

**Pull request closed without merging**